docs/oppen.md

```diff
@@ -0,0 +1,82 @@
+# Derek C. Oppen's pretty pritning algorithm
+
+Derek C. Oppen's pretty printing algorithm is a method for formatting code to make it more readable and visually appealing. It arranges code with indentation and line breaks. Oppen's algorithm has influenced many tools for code formatting and remains an important technique in software development for improving code clarity.
+
```

Comment:
That's nice for an academic paper, but we should sell differently:

1. What is the problem we're trying to solve
2. What are our assumptions/constraints?

Will need a call.